marklogic-data-hub/build.gradle

```diff
@@ -62,7 +62,7 @@ dependencies {
 
     compile 'commons-io:commons-io:2.4'
     compile 'org.apache.commons:commons-text:1.1'
-
+    compile 'ch.qos.logback:logback-classic:1.2.3'
```

Comment:
We don't want to include logback as a compile dependency, as that can cause issues for applications that depend on the DHF library as a dependency (it's not considered good form for libraries to include logging implementations either). 

I found this link for Spring Boot 1.x that shows a mechanism for creating a custom Gradle configuration that excludes a logging dependency - https://docs.spring.io/spring-boot/docs/1.2.0.M1/reference/html/build-tool-plugins-gradle-plugin.html#build-tool-plugins-gradle-repackage-custom-configuration 

I didn't find something similar in the Spring Boot 2.1.x docs though - https://docs.spring.io/spring-boot/docs/2.1.18.RELEASE/reference/html/build-tool-plugins-gradle-plugin.html 

Check for something like this though. Also test out the client JAR, which expects logging to work and in theory would have the same issue.

Reply:
I tried this locally, it seems to do the trick - be sure to test out both the client JAR and installer JAR:

```
configurations {
    datahubLogging
}

dependencies {
  ...
  datahubLogging 'ch.qos.logback:logback-classic:1.2.3'
}

// add this to both clientJar and bootJar
from(configurations.datahubLogging.collect { it.isDirectory() ? it : zipTree(it) }) {
        exclude "META-INF/*.SF"
        exclude "META-INF/*.DSA"
        exclude "META-INF/*.RSA"
    }
```

Reply:
It works fine for both client and installer jars. Thanks for the info on configuration. I will check it out.